packages/ai/src/generate-image/generate-image.ts

```diff
@@ -192,10 +187,23 @@ Only applicable for HTTP-based providers.
       for (const [providerName, metadata] of Object.entries<{
         images: unknown;
       }>(result.providerMetadata)) {
-        providerMetadata[providerName] ??= { images: [] };
```

Comment:
I guess I'm not sure why the existing code was written this way. Maybe @gr2m has more background -- ideally we would know why it was the way it was before making a change.

I can see the rework has compartmentalized the behavior change to the `gateway` provider.

Side note: Even apart from our changes, this seems like an expensive memory copy from the result `providerMetadata` to a new `providerMetadata` object. Images can be 4 - 10M or so. Avoiding any copy would be desirable if possible though not a showstopper.

Reply:
I expect the issue is not specific to gateway per se (and thus rohan's original edits) -- if a provider did happen to return other metadata it would probably get thrown away since there is not merging logic. But this may be theoretical as it is not clear any other provider actually does this today.